src/lib.rs

```diff
@@ -734,6 +739,36 @@ impl<T> IntoIterator for NonEmpty<T> {
     }
 }
 
+#[cfg(feature = "serialize")]
+pub mod serialize {
+    use std::{convert::TryFrom, fmt};
+
+    use super::NonEmpty;
+
+    #[derive(Debug)]
+    pub enum Error {
+        EmptyVec,
+    }
+
+    impl fmt::Display for Error {
+        fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+            match self {
+                Self::EmptyVec => f.write_str(
```

Comment:
Can we call this error `Empty`? I was confused at first and thought this is a new type of vector that is always empty :joy:

Reply:
Hahaha ya for sure :ok_hand: